src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -214,17 +317,131 @@ g.test('copy_contents_from_canvas')
 
     // Construct expected value for different dst color format
     const dstBytesPerPixel = kTextureFormatInfo[dstColorFormat].bytesPerBlock;
+    const format: RegularTextureFormat =
+      dstColorFormat === 'rgba8unorm-srgb' || dstColorFormat === 'bgra8unorm-srgb'
+        ? dstColorFormat === 'rgba8unorm-srgb'
+          ? 'rgba8unorm'
+          : 'bgra8unorm'
+        : dstColorFormat;
+    dstColorFormat;
```

Comment:
Last line appears to be a no-op? And the rest can be a bit more straightforward.

Since this is also used below, factor it out into a small helper function that documents the intent.
```ts
/**
 * If the destination format specifies a transfer function,
 * copyExternalImageToTexture (like B2T/T2T copies) should ignore it.
 */
function formatForExpectedPixels(format: RegularTextureFormat): RegularTextureFormat {
  return dstColorFormat === 'rgba8unorm-srgb'
    ? 'rgba8unorm'
    : dstColorFormat === 'bgra8unorm-srgb'
    ? 'bgra8unorm'
    : dstColorFormat;
}
```